### PR TITLE
Refactor, cleanup and add phonenumber & email

### PR DESCRIPTION
**Pull request by jrasanen**
- Added phone and email fields
- Change to sha256, as md5 is deprecated
- Clean the code to comply psr2
- Docker for tests & examples

### CODE REVIEW

**Review by jAAk3 — APPROVED** [review]

File: src/CheckoutFinland/Response.php [hunk 1]
```diff
@@ -63,14 +63,13 @@ public function validate()
     {
         $expected_mac = null;
 
-        if($this->algorithm == 3) {
-            $expected_mac = strtoupper(hash_hmac("sha256","$this->version&$this->stamp&$this->reference&$this->payment&$this->status&$this->algorithm", $this->merchant_secret));
-        }
-        else {
+        if ($this->algorithm == 3) {
+            $expected_mac = strtoupper(hash_hmac("sha256", "$this->version&$this->stamp&$this->reference&$this->payment&$this->status&$this->algorithm", $this->merchant_secret));
```
Comment (vesse) [inline src/CheckoutFinland/Response.php:67]: This does not look good. Works though.

File: tests/Spec/CheckoutFinland/PaymentSpec.php [hunk 2]
```diff
@@ -154,41 +155,40 @@ function it_calculates_a_mac_from_variables()
...
         $this->setOrderData(
-            $payment_data['stamp'], 
-            $payment_data['amount'], 
-            $payment_data['reference'], 
-            $payment_data['message'], 
+            $payment_data['stamp'],
+            $payment_data['amount'],
+            $payment_data['reference'],
+            $payment_data['message'],
             new \DateTime('2014-10-05')
```
Comment (vesse) [inline tests/Spec/CheckoutFinland/PaymentSpec.php:177]: Datetime here? Was string in Payment.php

**Review by vesse — APPROVED** [review]

File: tests/Spec/CheckoutFinland/PaymentSpec.php (hunk 2 above)
Comment (jrasanen) [reply tests/Spec/CheckoutFinland/PaymentSpec.php:177]: setDeliveryDate() casts it to a string in the format we require it to be

File: src/CheckoutFinland/Response.php (hunk 1 above)
Comment (jrasanen) [reply src/CheckoutFinland/Response.php:67]: Quite cringy, true.

**Pull request merged**